internal/service/device.go

```diff
@@ -263,10 +265,14 @@ func (h *ServiceHandler) ReplaceDeviceStatus(ctx context.Context, name string, d
 		device.Status.Lifecycle.Status = oldDevice.Status.Lifecycle.Status
 	}
 	oldDevice.Status = device.Status
-	common.UpdateServiceSideStatus(ctx, h.store, h.log, orgId, oldDevice)
+	changed := common.UpdateServiceSideStatus(ctx, h.store, h.log, orgId, oldDevice)
 
 	result, err := h.store.Device().UpdateStatus(ctx, orgId, oldDevice)
-	return result, StoreErrorToApiStatus(err, false, api.DeviceKind, &name)
+	status := StoreErrorToApiStatus(err, false, api.DeviceKind, &name)
+	if changed {
+		h.CreateEvent(ctx, GetResourceCreatedOrUpdatedEvent(ctx, false, api.DeviceKind, name, status, &api.ResourceUpdatedDetails{}))
+	}
+	return result, status
```

Comment:
Don't emit a generic "updated" event for status updates - emit specific events for certain status updates.  For example, changes in application status, CPU/RAM/disk thresholds, etc.